internal/controller/controllers/crd_utils.go

```diff
@@ -0,0 +1,69 @@
+package controllers
+
+import (
+	"context"
+
+	adiiov1alpha1 "github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const APIVersion = "adi.io.my.domain/v1alpha1"
+
+type CRDUtils struct {
+	client client.Client
+}
+
+func NewCRDUtils(client client.Client) *CRDUtils {
+	return &CRDUtils{client: client}
+}
+
+func (u *CRDUtils) CreateAgentCR(ctx context.Context, hostId, clusterNamespace, clusterName string) error {
+
+	if clusterNamespace == "" {
+		return nil
```

Comment:
When would this be the case, and what is the thinking behind returning silently returning?

Reply:
In case that the Cluster was not created via CRD.

Reply:
I see. I think that's worth adding as a comment on the method itself, also explaining what this method does and how it's intended to be ued.

Reply:
Done.